packages/gasket-plugin-middleware/lib/utils.js

```diff
@@ -89,14 +82,8 @@ async function executeMiddlewareLifecycle(gasket, app, middlewarePattern) {
   const middlewareConfig = config.middleware || [];
   const middlewares = [];
 
-  if (canUseMiddleware(app)) {
-    applyCookieParser(app, middlewarePattern);
-    applyCompression(app, config.compression);
-  } else {
-    debug('Skipping Express-specific middleware: app is not Express');
-  }
-
   await gasket.execApply('middleware', async (plugin, handler) => {
+    // @ts-ignore - TS is unable to infer that `app` can be either an Express application or a Fastify instance
```

Comment:
Why is that? Can't you make the type of `app` something like `ExpressApp | FastifyApp`?

Reply:
I didn't dive into the TS types. I just restored it back to how it was when it wasn't busted. We can look at them at a later date unless you want to contribute to the PR.